### DISCUSSION TITLE
Can Automerge sign with GPG-Keys?

### DISCUSSION THREAD

**Discussion by spammads** [renovatebot/renovate#23629] [Request Help] [answered]
### How are you running Renovate?

Placeholder value, please select the correct response from the dropdown

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I configured renovate to sign my commits with Gpg.
This works beautifully.
But automerged Commits dont get signed with the Gpg-Key.
Wondering if I missed a configuration option.
Thanks for helping & the brilliant tool.

### What you tried so far.

Search the documentation.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment]
depends on how you do the automerge. if you use platform automerge with PR then gitea needs a gpg key, as it's doing the merge and not renovate.

**Comment by spammads** [comment]
Ok. So if I set `platformAutomerge: true` and Gitea has a Gpg Key it will get signed?

**Reply by viceice** [reply]
it depends on gitea configuration

**Comment by spammads** [comment]
I already had set up Gitea to sign merges automatically.
I now changed my renovate-config to set platformAutomerge explicitly to true - though this should already be the default value according to documentation?!
Renovate now successfully merges my commits and signs them with its own Gpg-Key.
This is exactly the behaviour I wanted. 
But I dont get how the change of a (default-)config value telling Renovate to use Gitea as Merger can lead to this behavior.
I got a feeling i still misconfigured something under the hood...

**Reply by viceice** [reply, marked as answer]
before we had platform automerge renovate merged the PR itself and if you use merge commit, they are always only signed by platform. I generally use squash because we only have one commit anyways.

if you use rebase merge than only renovates commit is pushed to target branch, so you need to ensure that renovate also signs all commits.

**Reply by spammads** [reply]
I still dont fully grasp it but this is likely because of my weary head. Thanks for the assist. Best.

**Comment by johanneskastl** [comment]
Sorry for warming up this discussion.

I have enabled automerging for pre-commit changes in one repository.

```
    {
      "matchManagers": [
        "pre-commit"
      ],
      "automerge": true
    }
```

Renovate created a PR and merged it on the next run. But the merge commit was not signed, neither by Renovate (which uses a GPG key to sign its commits) nor by Forgejo (which also has a GPG key configured, which is used for signing the merge commit when manually clicking `Merge` in the WebUI).

There are no checks currently being run in this repository, but I did not yet include the `ignoreTests` setting (just found it).

Any idea why the merge commit was not signed? Should I disable `platformMerge`?

Kind Regards,
Johannes

**Reply by jamietanna** [reply]
@johanneskastl please raise a fresh Discussion
